storagenode/peer.go

```diff
@@ -0,0 +1,211 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package storagenode
+
+import (
+	"context"
+	"net"
+
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/grpc"
+
+	"storj.io/storj/pkg/identity"
+	"storj.io/storj/pkg/kademlia"
+	"storj.io/storj/pkg/node"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/piecestore/psserver"
+	"storj.io/storj/pkg/piecestore/psserver/psdb"
+	"storj.io/storj/pkg/provider"
+	"storj.io/storj/pkg/server"
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/storage"
+)
+
+// DB is the master database for Storage Node
+type DB interface {
+	// TODO: use better interfaces
+	Disk() string
+	PSDB() *psdb.DB
+	RoutingTable() (kdb, ndb storage.KeyValueStore)
+}
+
+// Config is all the configuration parameters for a Storage Node
+type Config struct {
+	Identity identity.Config
+
+	// TODO: switch to using server.Config when Identity has been removed from it
+	PublicAddress string `help:"public address to listen on" default:":7777"`
+	Kademlia      kademlia.Config
+	Piecestore    psserver.Config
```

Comment:
For sure `Piecestore`? currently we are using `Storage`. Or my understanding wrong and this won't be used with cmd later.

Reply:
Yeah, storage makes more sense

Reply:
Fixed